### PR TITLE
[styling] Refactor codebase frontend

### PR DESCRIPTION
**Pull request by selene-huang**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

This PR performs an overhaul of the codebase to make the frontend cleaner and more legible. Primarily, I attempted to remove unnecessary containers, reuse components where possible, and standardize organizational and naming schemes. Some screens' margins may have been altered during the switch to reusing standardized components, but other than that the frontend should retain its original styling.

**Codebase-Wide Changes**
- Changed all occurrences of transactions to invoices to reflect the updated name
- Changed all occurrences of voucher range to voucher types to differentiate the object from the new input format
- Created new firebase collections `voucher-types` and `invoices` (as renaming an old collection is impossible)
  - Note: while the old firebase collection `voucher-ranges` has been deleted, `transactions` remains for testing purposes
- Updated all percentages to fixed pixel values EXCEPT:
  - any occurrences of width = 100%
  - SortModalContainer (height = 55%)
  - FilterModalContainer (height = 75%)
  - NavButtonBase (width = 33.333%)
  - StartContainer (top = 25%)
- Changed all interfaces to types for consistency's sake (further reading: [[1]](https://stackoverflow.com/questions/37233735/interfaces-vs-types-in-typescript) [[2]](https://medium.com/@martin_hotell/interface-vs-type-alias-in-typescript-2-7-2a8f1777af4c))
- Changed inline function declarations into named functions ([further reading](https://stackoverflow.com/questions/50350202/when-to-use-inline-function-on-button-onclick-event-javascript-react-js))
- Changed all instances of 'white' and '#fff' to 'Colors.offWhite' for consistency
- Moved constants that don't depend on screen values outside of the component function
- Attempted to standardize function, component, and prop names

**Component Updates**
- Replaced react native SafeAreaView, which is IOS only, with [expo's SafeAreaView](https://docs.expo.dev/versions/latest/sdk/safe-area-context/) 
- NavButton refactor:
  - The old NavButton component has been renamed to NavButtonBase
  - Removed styles.selectedBtn and styles.unselectedBtn; instead, NavButtonBase now takes in a prop isSelected
  - Created a new component, NavButton, to clean up repeated code in VoucherEntryNavigator
- Moved common code from StatusComponentRed & StatusComponentGreen into StatusComponentBase
- Moved common code from ButtonMagenta and ButtonWhite into ButtonBase
- Merged SelectedFilterField and UnselectedFilterField into one component, FilterFieldBase, that takes in a prop isSelected
- Merged SelectedFilterTag and UnselectedFilterTag into one component, FilterTagBase, that takes in a prop isSelected
- Merged SelectedRadioButton and UnselectedRadioButton into one component, RadioButtonBase, thattakes in a prop isSelected
- Merged SelectedSortAndFilterBase and UnselectedSortAndFilterBase into one component, SortAndFilterBase, that takes in a prop isSelected
- Removed instances of duplicate components (ex: LeftAlignContainer in the profiles flow and LeftAlignContainer in the invoices flow)
- Removed unused components (ex: AddManuallyButton)

**Organizational Changes**
- Renamed OneLine to Row and renamed Row to Card to better reflect their usages
- Renamed RadioButton in `src/components/invoices` to RadioButtonList to emphasize the fact that it is a list of radio buttons
- Moved any assets that were used across multiple flows to `assets/Components`
- Moved containers used for styling in `src/components/common` into their respective component file and removed the file `src/components/common/styling`
- Moved InputField and StartLogo to auth, because they are only used in the auth flow
- Moved RadioButton to invoices because it is only used in the invoices flow
- Reorganized `src/components/invoices/styles.ts` so that components used in the same file are grouped together

**Miscellaneous Changes**
- Replaced hardcoded tab values in VoucherEntryNavigator screen with an enum `tabs`
- Merged showStartInvalidError and showEndInvalidError in the VoucherBatch screen into showErrorMessage
- Standardized frontend for error messages such that all screens that need to conditionally render an error message now contain the useState variables `showErrorMessage` and `errorMessage`

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"

The old versions of screens are on the left, new versions on the right. Please keep in mind that my screenshots will all be shifted down slightly since I switched the SafeAreaView to one that works with Android.

`StartScreen.tsx`
 

`LoginScreen.tsx`
 

`ForgotPasswordScreen.tsx`
 

`VoucherEntryStartScreen.tsx`
 

`ManualVoucherScreen.tsx`
 

`ConfirmValueScreen.tsx`
 

`VoucherBatchScreen.tsx`
 

`ScanningScreen.tsx`
 

`ReviewScreen.tsx` 
 

`ConfirmationScreen.tsx` 
 

`ProfileScreen.tsx` 
 

`ContactUsScreen.tsx` 
 

`InvoicesScreen.tsx` 
 

Invoices screen + Filter Modal 
 

Invoices screen + Sort Modal 
 

`InvoiceDetailsScreen.tsx` 
 

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"

Run `npx expo start` and click through every screen to make sure that styling and alignment looks good, and that everything still functions as intended.

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

- Update the Firebase transactions collection.
- Consider updating the scanning flow's screen names

### Online sources
[//]: # "Optional - copy links to any tutorials or documentation that was useful to you when working on this PR"

- https://www.perssondennis.com/articles/react-anti-patterns-and-best-practices-dos-and-donts
- https://github.com/styled-components/styled-components/issues/630#issuecomment-317277803
- https://stackoverflow.com/questions/39336266/react-native-inline-styles-and-performance

[//]: # "This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation."
🥦 CC: @sauhardjain

### CODE REVIEW

File: src/database/queries.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ const testColl = collection(db, 'test-col');
 const vendorCollection = collection(db, 'vendors');
 const voucherRangeCollection = collection(db, 'voucher-ranges');
 const voucherCollection = collection(db, 'vouchers');
-const transactionCollection = collection(db, 'transactions');
+const invoiceCollection = collection(db, 'transactions');
```
Comment (ronniebeggs) [inline src/database/queries.ts]: The firebase collection is still called 'transactions' and not invoices. Is there any reason for not changing this?

File: src/database/queries.ts [hunk 2]
```diff
@@ -297,21 +297,21 @@ export const getVouchersByVendorUuid = async (
...
-    const dbQuery = query(transactionCollection, where('uuid', '==', uuid));
+    const dbQuery = query(invoiceCollection, where('uuid', '==', uuid));
     const querySnapshot = await getDocs(dbQuery);
-    return querySnapshot.docs[0]?.data() as Transaction;
+    return querySnapshot.docs[0]?.data() as Invoice;
   } catch (e) {
     // eslint-disable-next-line no-console
-    console.warn('(getTransaction)', e);
+    console.warn('(getInvoice)', e);
     throw e;
```
Comment (ronniebeggs) [inline src/database/queries.ts:311]: Is there a cap on the number of vouchers able to be added to an invoice? I noticed that the $5100 invoice only had 51 vouchers. Was the value of the invoice manually inputted in the database, or was there some sort of error here?
Screen Shot 2023-05-15 at 3 35 32 PM

File: src/database/queries.ts [hunk 3]
```diff
@@ -32,7 +32,7 @@ const testColl = collection(db, 'test-col');
 const vendorCollection = collection(db, 'vendors');
 const voucherRangeCollection = collection(db, 'voucher-ranges');
```
Comment (ronniebeggs) [inline src/database/queries.ts]: Can we rename 'voucher-range' to 'voucher-type' or something similar? It gets confusing especially now that we have the functionality for inputting a range of vouchers. If, for whatever reason the 'voucher-range' must stick, maybe we change everything related to range inputs to 'voucher-batch'.

File: src/screens/scanning/ConfirmValueScreen.tsx [hunk 4]
```diff
@@ -97,53 +93,44 @@ export default function ConfirmValueScreen({
...
+          separator="."
+          delimiter=","
+          precision={2}
+        />
+        <ErrorContainer>
+          <Body2Subtext>
+            <RedText>
+              {showZeroError && 'Voucher must be redeemed for more than $0.'}
+              {showExceedError &&
+                'Voucher value exceeds maximum redemption limit.'}
```
Comment (ronniebeggs) [inline src/screens/scanning/ConfirmValueScreen.tsx]: Possibly standardize process for displaying error messages. This screen has separate state variables for each error type, while others have one 'showInvalidError,' where the error message to be displayed is another state variable. Not sure which one is better, but it'd be good to be consistent.

File: src/screens/scanning/VoucherEntryNavigator.tsx [hunk 5]
```diff
@@ -6,124 +6,96 @@ import {
...
-              handlePreventLeave({
-                hasUnsavedChanges,
-                navigation,
-                dispatch,
-              })
-            }
-          >
+        <StandardHeader>
+          <H2Heading>Add a voucher</H2Heading>
+          <TouchableOpacity onPress={onPreventLeave}>
```
Comment (ronniebeggs) [inline src/screens/scanning/VoucherEntryNavigator.tsx:72]: Don't know if this is an issue introduced in this pr, but there isn't much spacing above the top bar header. Especially with the notch on iPhones 10-13, the header rubs against the top of the screen.
IMG_EE17B4F39994-1

File: src/database/queries.ts (hunk 1 above)
Comment (selene-huang) [reply src/database/queries.ts]: Didn't wanna mess with the backend without Sauhard greenlighting things LOL @sauhardjain should we change that?

File: src/screens/scanning/VoucherEntryNavigator.tsx (hunk 5 above)
Comment (selene-huang) [reply src/screens/scanning/VoucherEntryNavigator.tsx:72]: I don't think I changed the header spacing so the only thing I can think of would be that the SafeAreaView is no longer working as intended. If you replace the `SafeArea` with a `SafeAreaView` imported from react native, does this solve the issue?

File: src/screens/scanning/ConfirmValueScreen.tsx (hunk 4 above)
Comment (selene-huang) [reply src/screens/scanning/ConfirmValueScreen.tsx]: Good point, I'll refactor them to all follow the showError + errorMessage state variable format

**Review by sauhardjain — APPROVED** [review]
lgtm!
